src/sentry/integrations/slack/utils/channel.py

```diff
@@ -188,26 +198,68 @@ def get_channel_id_with_timeout(
     return prefix, channel_id, False
 
 
-def check_for_channel(client: SlackClient, name: str) -> str | None:
-    msg_response = client.post(
-        "/chat.scheduleMessage",
-        data={
-            "channel": name,
-            "text": "Sentry is verifying your channel is accessible for sending you alert rule notifications",
-            "post_at": int(time.time() + 500),
-        },
-    )
+def check_for_channel(
+    client: SlackClient | SlackSdkClient,
+    name: str,
+) -> str | None:
+
+    if isinstance(client, SlackSdkClient):
+        try:
+            msg_response = client.chat_scheduleMessage(
+                channel=name,
+                text="Sentry is verifying your channel is accessible for sending you alert rule notifications",
+                post_at=int(time.time() + 500),
+            )
+        except SlackApiError as e:
+            logger_params = {
+                "error": str(e),
+                "integration_id": client.integration_id,
+                "channel": name,
+                "post_at": int(time.time() + 500),
+            }
+            logger.exception("slack.chat_scheduleMessage.error", extra=logger_params)
+            raise
+
+        if "channel" not in msg_response:
+            return None
+
+        try:
+            client.chat_deleteScheduledMessage(
+                channel=msg_response["channel"],
+                scheduled_message_id=msg_response["scheduled_message_id"],
```

Comment:
```suggestion
                channel=msg_response.get("channel"),
                scheduled_message_id=msg_response.get("scheduled_message_id"),
```
the `SlackResponse` class has a built in method called `get` that fetches the information from `response.data`